ibm/resource_ibm_private_dns_glb_pool_test.go

```diff
@@ -0,0 +1,148 @@
+					testAccCheckIBMGlbPoolExists("ibm_dns_glb_pool.test-pdns-pool-nw", resultprivatedns),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMPrivateDNSGlbPoolImport(t *testing.T) {
+	var resultprivatedns string
+	name := fmt.Sprintf("testpdnszone%s.com", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPrivateDNSGlbPoolDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMGlbPoolBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMGlbPoolExists("ibm_dns_glb_pool.test-pdns-pool-nw", resultprivatedns),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "ibm_dns_glb_pool.test-pdns-pool-nw",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckIBMGlbPoolBasic(name string) string {
+	return fmt.Sprintf(`
+	data "ibm_resource_group" "rg" {
+		name = "default"
+	}
+	resource "ibm_is_vpc" "test-pdns-glb-pool" {
+		name = "test-pdns-glb-pool"
+		resource_group = data.ibm_resource_group.rg.id
+	}
+	resource "ibm_resource_instance" "test-pdns-glb-pool-instance" {
+		name = "test-pdns-glb-pool-instance"
+		resource_group_id = data.ibm_resource_group.rg.id
+		location = "global"
+		service = "dns-svcs"
+		plan = "standard-dns"
+	}
+	resource "ibm_dns_zone" "test-pdns-glb-pool-zone" {
+		name = "%s"
+		instance_id = ibm_resource_instance.test-pdns-glb-pool-instance.guid
+		description = "testdescription"
+		label = "testlabel"
+	}
+	resource "ibm_dns_glb_pool" "test-pdns-pool-nw" {
+		depends_on = [ibm_dns_zone.test-pdns-glb-pool-zone]
+		name = "testpool"
+		instance_id = ibm_resource_instance.test-pdns-glb-pool-instance.guid
+		description = "new test pool"
+		enabled="true"
+		healthy_origins_threshold=1
+		origins {
+				name    = "example-1"
+				address = "www.google.com"
+				enabled = true
+				description="origin pool"
+		}
+    }
+	  `, name)
+}
+
+func testAccCheckIBMPrivateDNSGlbPoolDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_dns_glb_pool" {
+			continue
+		}
+
+		pdnsClient, err := testAccProvider.Meta().(ClientSession).PrivateDnsClientSession()
+		if err != nil {
+			return err
+		}
+
+		parts := rs.Primary.ID
+		partslist := strings.Split(parts, "/")
+
+		getGlbPoolOptions := pdnsClient.NewGetPoolOptions(partslist[0], partslist[1])
+		_, res, err := pdnsClient.GetPool(getGlbPoolOptions)
+
+		if err != nil &&
+			res.StatusCode != 403 &&
+			!strings.Contains(err.Error(), "The service instance was disabled, any access is not allowed.") {
+
+			return fmt.Errorf("testAccCheckIBMPrivateDNSGlbPoolDestroy: Error checking if instance (%s) has been destroyed: %s", rs.Primary.ID, err)
+		}
+	}
+	return nil
+}
+
+func testAccCheckIBMGlbPoolExists(n string, result string) resource.TestCheckFunc {
```

Comment:
check other test case and modify the code

Reply:
done